lib-sql/functions/place_triggers.sql

```diff
@@ -92,6 +92,16 @@ BEGIN
     -- Get the existing place_id
     select * from placex where osm_type = NEW.osm_type and osm_id = NEW.osm_id and class = NEW.class and type = NEW.type INTO existingplacex;
 
+    -- Pure postcodes are never queried from placex so we don't add them.
+    -- location_postcodes is filled from the place table directly.
+    IF NEW.class = 'place' AND NEW.type = 'postcode' THEN
+      -- Remove old placex entry if the type changed to postcode.
+      IF existingplacex.type IS NOT NULL AND existingplacex.type != 'postcode' THEN
```

Comment:
Let's leave out the second condition. Without it, an existing database can slowly 'clean out' itself.